php/stats.php

```diff
@@ -22,12 +24,12 @@
     $sth->execute([$trid]);
     $row = $sth->fetch();
     if (!$row) {
-        die('Error;' . _("Trip not found."));
+        die(json_encode(["error" => ("Trip not found.")]));
```

Comment:
Localistion dropped? `_(` has become `(`; brackets aren't adding anything in the current form

Could potentially use `json_error()` from `helper.php`.

Reply:
I definitely need to do something here but I'm not trying to remove the translation. There is a `gettext` call for this string (for whatever is sent in the `error` prop in the UI now (line 2277). It works right now because we already have the string in the catalogue but the missing `_` is a problem. However, I don't want to do localization here in this file.

I definitely **don't** want to use `json_error` (and `json_success`) because, IMO, data endpoints should not send back localized messages. That's a presentation layer concern.

I'm thinking of defining `_` as the identity function in here so strings can be properly marked for localization in the UI.

Reply:
It would be better having message codes, rather than strings passed wholesale to the localisation functions...

But I think that gets a little harder, changes how the defaults are defined (and found in source code) and a whole other host of problems.

So changes the workflow in one way or another too.

>I'm thinking of defining `_` as the identity function in here so strings can be properly marked for localization in the UI.

Which is what `locale.php` does for `if !$OF_USE_LOCALES`... And as long as no other PHP file includes `stats.php`, shouldn't cause an issue.

At some point, these want to become proper classes (especially if we go down the route of using a proper router etc), but that's a concern for down the road.

These could return an error code (and an english string? Ideally...), and then have the mapping from code => message in the JS and gettext should pick them up from there instead. And do the localisation there too.

And/or forcing a locale of `en_us` in standalone scripts like this, so the localisation function just returns the string anyway.

Not that `json_success`/`json_error` do any sort of code, just a status number...